packages/browser/src/plugins/remote-loader/index.ts

```diff
@@ -100,7 +229,22 @@ export async function remoteLoader(
 
           validate(plugins)
 
-          allPlugins.push(...plugins)
+          const routing = routingRules.filter(
+            (rule) => rule.destinationName === remotePlugin.creationName
+          )
+
+          plugins.forEach((plugin) => {
+            const wrapper = new ActionDestination(
+              remotePlugin.creationName,
+              plugin
+            )
+
+            if (routing.length && routingMiddleware) {
```

Comment:
I think we should also check if the plugin is `destination` type. That way we don't do any filtering on cloud-mode actions that have a web component (like amplitude/braze cloud)